### DISCUSSION TITLE
.Net MAUI code adaptation

### DISCUSSION THREAD

**Discussion by robertodalmonte** [TheCodeTraveler/AsyncAwaitBestPractices#99] [Q&A] [answered]
Hi Brandon,
I'm trying to use your fantastic AsyncAwaitBestPractices and I made some minor changes to make it work in a .Net MAUI project.
HackerNews/ViewModels/Base
BaseViewModel.cs
   line 77 and 78: (Xamarin.Forms.)Application.Current becomes:
                           Microsoft.Maui.Controls.Application.Current
                           
                           I gues that line 78 should read
                           await MainThread.InvokeOnMainThreadAsync(() => Application.Current.MainPage.IsBusy = isBusy);
                           instead of
                           await MainThread.InvokeOnMainThreadAsync(() => Application.Current.MainPage.IsBusy = true);
                           
   line 50:            await using var stream = await Client.GetStreamAsync(apiUrl).ConfigureAwait(false);

I also made GetDataObjectFromApi and UpdateActivityIndicatorStatus methods static.

Since my knowledge of AsyncAwait practices is mediocre (at best), I was wondering if those modifications make sense.
Secondly, since I'm using Resharper, I get the following warning:
_networkIndicatorCount
Suspicious "volatile" field usage: compound operation is not atomic. "Interlocked" class can be used instead.
As you might guess, I have no idea what the heck they're talking about...

What's your take on this?
Best regards
Roberto

**Comment by TheCodeTraveler** [comment, marked as answer]
Hey @robertodalmonte! 

I've updated the HackerNews sample to .NET MAUI in its own repo: https://github.com/brminnick/hackernews. Feel free to check it out and compare/contrast your own code!

**Reply by TheCodeTraveler** [reply]
I'll update the HackerNews sample app in this repo once .NET MAUI GAs in [Q2 2022](https://github.com/dotnet/maui/wiki/Roadmap)

**Reply by TheCodeTraveler** [reply]
Don't worry too much about Resharper's recommendation, to update `static volatile int _networkIndicatorCount`.

They're basically saying that I should use [`lock`](https://docs.microsoft.comdotnet/csharp/language-reference/keywords/lock-statement?WT.mc_id=mobile-0000-bramin) instead of `volatile`. 

It's not a bad recommendation, but it'll force me to add some extra code. And since I use this repo for teaching introductory concepts, I prefer to keep the code more straightforward and simple.
